crates/bitcoin/src/parser.rs

```diff
@@ -404,12 +409,28 @@ fn parse_transaction_output(raw_output: &[u8]) -> Result<(TransactionOutput, usi
     ))
 }
 
+pub(crate) fn extract_address_hash_witness(witness_script: &[u8]) -> Result<Address, Error> {
+    if witness_script.len() == 33 {
+        // compressed public key format
+        let prefix = *witness_script.get(0).ok_or(Error::EndOfFile)?;
+        if prefix == SECP256K1_TAG_PUBKEY_EVEN || prefix == SECP256K1_TAG_PUBKEY_ODD {
```

Comment:
Maybe you can add a link that describes the format? Otherwise it's quite difficult to assess the correctness of this code (for me, but also for the auditors)

Reply:
Link to C code above constants, added short description on format.